### PR TITLE
ENH progressbars for addurl

### PR DESCRIPTION
**Pull request by yarikoptic**
This pull request fixes https://github.com/datalad/datalad-container/issues/49

- It got even messier in there, and ideally requires some fixups on git-annex side: http://git-annex.branchable.com/todo/provide___39__file__39___in_--json-progress_record_for_addurl/
- I also disabled Total progress bar if we expect only one .  hopefully I didn't ruin something great ;-)

I guess would require manual testdriving since it is UI which we might not test very thoroughly automatically.  But on my attempts seems to be quite kosher.

### CODE REVIEW

File: datalad/support/annexrepo.py [hunk 1]
```diff
@@ -2057,7 +2057,10 @@ def add_url_to_file(self, file_, url, options=None, backend=None,
             out_json = self._run_annex_command_json(
                 'addurl',
                 opts=options + [files_opt] + [url],
-                log_online=True, log_stderr=False,
+                progress=True,
+                expected_entries={file_: None},
+                #log_online=True,
+                #log_stderr=False,
```
Comment (kyleam) [inline datalad/support/annexrepo.py]: Why leave commented out code?

File: datalad/support/annexrepo.py [hunk 2]
```diff
@@ -2279,6 +2283,8 @@ def _run_annex_command_json(self, command,
         expected_entries : dict, optional
           If provided `filename/key: size` dictionary, will be used to create
           ProcessAnnexProgressIndicators to display progress
+        progress: bool, optional
+          Either request/handle --json-progress
```
Comment (kyleam) [inline datalad/support/annexrepo.py]: s/Either/Whether to/

File: datalad/support/annexrepo.py [hunk 3]
```diff
@@ -3650,8 +3676,8 @@ def __call__(self, line):
 
         # so we have a progress indicator, let's deal with it
         action = j['action']
-        download_item = action.get('file') or action.get('key')
-        download_id = (action['command'], action['key'])
+        download_item = action.get('key') or action.get('file')
```
Comment (kyleam) [inline datalad/support/annexrepo.py]: "download_item" seems like a weird name at the level of `ProcessAnnexProgressIndicators`.  Regardless of where this is used at the moment, couldn't the item correspond to  an upload or a local copy or any other action on a file/key?

Comment (yarikoptic) [reply datalad/support/annexrepo.py]: indeed!  I guess it i more of an `action_item`, will change now. thanks

File: datalad/support/annexrepo.py (hunk 1 above)
Comment (yarikoptic) [reply datalad/support/annexrepo.py]: dunno... dirty yarik. will remove

**Comment by yarikoptic** [comment]
I've restarted a travis build (hopefully uploaded fresh git-annex resolves the issue) since failure looked unrelated.  I am afraid I did break something with this PR since my checks I had to add hint that something is behaving differently now.  But I am afraid that due to this "interactive" nature of the element here, we would only discover by trying it out.
I will merge tomorrow if no new problems emerge

**Pull request merged**